### ISSUE TITLE
JUnit 5 support 

### ISSUE THREAD

**Issue by joecks** [robolectric/robolectric#3477]
Currently there is no way to run JUnit 5 tests with robolectric. Are there any plans to introduce it or could someone sketch what would be needed for an integration so I could contribute the code.

**Comment by bangarharshit** [comment]
Duplicate issue - https://github.com/robolectric/robolectric/issues/3397.

**Comment by alixwar** [comment]
@bangarharshit I took the liberty to rename #3397 to scope that issue to  make Robolectric tests run in legacy mode (tests run by the JUnit5 vintage runner) under JUnit5. I think this issue is about "real" JUnit5 support (tests run by the JUnit5 jupiter runner)

**Comment by technoir3** [comment]
FYI Gradle 4.6 is out which supports JUnit Platform out of the box so we can now run JUnit 5 tests in Android projects without any 3rd party plugins.
It'd be great to have a JUnit 5 extension for Robolectric so we could use the awesome new `@Nested` and `@ParametrizedTest`.

**Comment by CoderSpinoza** [comment]
@alixwar I agree that this issue and #3397 should be separated. :) Is there a milestone set for this issue? Really wanna try out junit 5 (jupiter) with Robolectric in my project.

**Comment by xian** [comment]
https://junit.org/junit5/docs/current/user-guide/#extensions

**Comment by GaganBhat** [comment]
Just following up, is there a way to run JUnit 5 tests with robolectric now? I see some merges occurred with JUnit5 support.

**Comment by BergnerArn** [comment]
I've tried to enhance the test instance creation with the Roboeletric Class Loading process in order to make the Android OS available via JUnit5 Extension Point. Fact is that Jupiter detects the difference of the instance. Other PoC failed as well since the migration is a very complex problem.

I now understand that migrating this complete framework to JUnit5 will be a lot of pain. But since JUnit4 is very old, it should be considered in order to decrease tech dept. Even the performance of Jupiter is a good argument.

**Comment by Anish528** [comment]
Hi @utzcoz 
I would like to explore this project JUnit 5 Support as part of the GSoC program.
Any advice or tips on how to get started would really help out!
Thanks.

**Comment by utzcoz** [comment]
@Anish528 Thanks for interesting for this project. Could you create a new discussion at GitHunb Discussion as other folks did for GSoC 2022? We can talk more about it before applying proceas without breaking discussion on this issue.

**Comment by TWiStErRob** [comment]
Please link here.

**Comment by utzcoz** [comment]
There are three existing discussions about GSoC 2022 program: https://github.com/robolectric/robolectric/discussions/7141, https://github.com/robolectric/robolectric/discussions/7127 and https://github.com/robolectric/robolectric/discussions/7128.

**Comment by Anish528** [comment]
> @Anish528 Thanks for interesting for this project. Could you create a new discussion at GitHub Discussion as other folks did for GSoC 2022? We can talk more about it before applying proceas without breaking discussion on this issue.

Sure will do Thanks!

**Comment by VarenytsiaMykhailo** [comment]
already 2023, is there any progress with the supporting of JUnit 5, it is already 6 years old?

**Comment by hoisie** [comment]
Nope, unfortunately there has been no progress on this issue, and I am not aware of anyone working on this at the moment. For several years now there has been a project on the backburner to decouple Robolectric from JUnit4, but that has not happened yet.

**Comment by trevor-hackman-kr** [comment]
6 years later, why isn't this fixed yet?

**Comment by MuhammadKhoshnaw** [comment]
Is there any update? I don't wanna add junit4 to my project just for robolectric 😞

**Comment by simvn** [comment]
Any update or outlook on when we might expect a migration?

**Comment by victorpineda-jt** [comment]
Same issue here, 2024, happy new year!

**Comment by emartynov** [comment]
Can you run Robolectric with JUnit5 vintage engine?

**Comment by TWiStErRob** [comment]
As far as I know, yes, but it's provided as [a temporary migration path](https://junit.org/junit5/docs/current/user-guide/#:~:text=JUnit%20provides,JUnit%20Vintage%20test%20engine).

**Comment by hoisie** [comment]
Just to be clear and set expectations:

While I personally think it would be great for Robolectric to have deep integration with JUnit5, this project is considered low priority and there is nobody I know actively working on it. I am personally focusing on other projects and initiatives. Unfortunately, for various reasons, JUnit5 is not widely used at Google.

If JUnit5 support for Robolectric is important for people, I would warmly encourage someone to take a more active role in investigating and developing a possible solution. I'd be happy to provide support for this.

**Comment by warnyul** [comment]
I've created an experimental project to explore JUnit 5 integration. This project have limitations, but it can serve as a foundation for further collaboration: https://github.com/apter-tech/junit5-robolectric-extension

**Comment by mannodermaus** [comment]
This is fantastic work, @warnyul – I'm able to confirm that this does indeed work for a simple use case that I just spun up. Great job!

**Comment by warnyul** [comment]
@utzcoz  You mean as it is?  I think in the Robolectric repo we could do more than what I did in my experiment.

I just realized there was a GSOC project about JUnit5 support, but I missed the deadline. Is it feasible to complete this as a part of the GSOC program? It would be beneficial to talk about the process of decoupling Robolectric from JUnit 4.

**Comment by TomTruyen** [comment]
@warnyul 's "experiment" works without problems so it would be great if it gets implemented in robolectric itself

**Comment by warnyul** [comment]
@TomTruyen I plan to create a PR with the changes. I couldn't take the time yet.

**Comment by jcshartz** [comment]
It's 2025 and this issue is still alive 😔

**Comment by mmirzynski** [comment]
@warnyul thank for your extension, it really helps!
Heya @utzcoz , just totally out of curiosity as I understand you're busy: are there any plans to add JUnit5 support in Robolectric in the future?

**Comment by utzcoz** [comment]
@mmirzynski Sorry for late response. The JUnit5 has been in the Robolectric plan some years ago, but it doesn't have a very high priority. We have tried to make some progress for it to add it to GSoC projects, but it's very difficult for students, and no-one selects it. It is in my personal plan for Robolectric too, but I don't have enough time for it because of my work. I think there is an approach might work: the community contributes code to the Robolectric to make it work for Robolectric. I can help to push the progress. As I know, there is a community project to make JUnit5 work with Robolectric at almost scenario, you can try it: https://github.com/apter-tech/junit5-robolectric-extension.

**Comment by jcshartz** [comment]
It's 2026 and they already released JUnit 6
